src/components/ui/Toggle/toggle.scss

```diff
@@ -0,0 +1,190 @@
+  --fs-toggle-checked-knob-bkg-color           : var(--fs-control-bkg-default);
+  --fs-toggle-checked-knob-border-color        : var(--fs-toggle-checked-knob-bkg-color);
+
+  --fs-toggle-hover-shadow                     : var(--fs-shadow-hover);
+  --fs-toggle-hover-bkg-color                  : var(--fs-color-primary-bkg-light);
+  --fs-toggle-hover-border-color               : var(--fs-border-color-default-hover);
+  --fs-toggle-hover-knob-bkg-color             : var(--fs-toggle-hover-border-color);
+  --fs-toggle-hover-knob-border-color          : var(--fs-toggle-hover-border-color);
+
+  --fs-toggle-disabled-bkg-color               : var(--fs-color-disabled-bkg);
+  --fs-toggle-disabled-border-color            : var(--fs-border-color-default-disabled);
+  --fs-toggle-disabled-knob-bkg-color          : var(--fs-color-neutral-5);
+  --fs-toggle-disabled-knob-border-color       : var(--fs-toggle-disabled-knob-bkg-color);
+
+  // --------------------------------------------------------
+  // Structural Styles
+  // --------------------------------------------------------
+
+  display: flex;
+  align-items: center;
+  padding: 0;
+  margin: 0;
+  border: 0;
+  outline: 0;
+
+  input {
+    @include focus-ring-visible;
+
+    position: absolute;
+    z-index: 1;
+    display: block;
+    width: 100%;
+    height: 100%;
+    cursor: pointer;
+    background-color: var(--fs-toggle-bkg-color);
+    border: var(--fs-toggle-border-width) solid var(--fs-toggle-border-color);
+    border-radius: var(--fs-toggle-border-radius);
+    box-shadow: var(--fs-toggle-shadow);
+    transition: var(--fs-toggle-transition-property) var(--fs-toggle-transition-timing) var(--fs-toggle-transition-function);
+    appearance: none;
+
+    @media (hover: hover) {
+      &:hover {
+        background-color: var(--fs-toggle-hover-bkg-color);
+        border-color: var(--fs-toggle-hover-border-color);
+        box-shadow: var(--fs-toggle-hover-shadow);
+
+        + [data-fs-toggle-knob] {
+          background-color: var(--fs-toggle-hover-knob-bkg-color);
+          border-color: var(--fs-toggle-hover-knob-border-color);
+
+          [data-fs-toggle-knob-icon] {
+            color: var(--fs-toggle-knob-icon-color);
+          }
+        }
+      }
+    }
+
+    &:checked {
+      background-color: var(--fs-toggle-checked-bkg-color);
+      border: var(--fs-toggle-border-width) solid var(--fs-toggle-checked-border-color);
+
+      + [data-fs-toggle-knob] {
+        background-color: var(--fs-toggle-checked-knob-bkg-color);
+        border-color: var(--fs-toggle-checked-knob-border-color);
+
+        [data-fs-toggle-knob-icon] {
+          color: var(--fs-toggle-knob-icon-checked-color);
+        }
+      }
+
+      @media (hover: hover) {
+        &:hover {
+          background-color: var(--fs-toggle-checked-hover-bkg-color);
+          border: var(--fs-toggle-border-width) solid var(--fs-toggle-checked-hover-border-color);
+        }
+      }
+    }
+
+    &:disabled {
```

Comment:
When it's disabled, should it really have this hover state?
![CleanShot 2022-04-27 at 09 49 29](https://user-images.githubusercontent.com/381395/165522087-4141fd7c-3f9d-438a-9c9b-e7960a5ec828.gif)

Reply:
nice catch, I deactivated the `input` pointer-events instead of the root container's